File: SmartBus.Website/Data/Entities/Route.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace SmartBus.Website.Data.Entities
{
    public class Route : IEntity
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public int IntervalInMinutes { get; set; }
        public DateTime FirstLeavingHour { get; set; }
        public DateTime LastLeavingHour { get; set; }
        public decimal Cost { get; set; }
        public ICollection<Trip> Trips { get; set; }
        public ICollection<Driver> Drivers { get; set; }
        public ICollection<Bus> Buses { get; set; }
    }
}

File: SmartBus.Website/ClientApp/src/app/models/route.ts
import { IDriver } from './driver';
import { IBus } from './bus';
import { IUser } from './user';

export interface IRoute {
  id: number;
  name: string;
  intervalInMinutes: number;
  firstLeavingHour: Date | string;
  lastLeavingHour: Date | string;
  drivers?: IDriverFromDb[];
  buses?: IBus[];
}

export interface IDriverFromDb {
  id: number;
  registerDate: Date | string;
  phone: string;
  userId: number;
  user: IUser;
  routeId?: number;
}

